api/specs.proto

```diff
@@ -128,6 +130,11 @@ message TaskSpec {
 	uint64 force_update = 9;
 }
 
+message CustomSpec {
+	string name = 1;
```

Comment:
I suppose this is describing which kind of spec is enclosed - i.e. `plugin`, `container` etc (perhaps with some `com.docker.` scheme) - not an arbitrary name. If so, maybe `kind` is a better name? I would also have suggested `type`, but that's a reserved word.